include/dnnl_types.h

```diff
@@ -2412,6 +2412,7 @@ typedef enum {
     /// Any ISA (excepting those listed as initial support)
     dnnl_cpu_isa_all = 0x0,
 
+#if DNNL_X64
```

Comment:
Currently we do not expose the architecture externally (through `dnnl_config.h` or `dnnl_types.h`).
This means that on the user side (other than in our examples and tests) neither `DNNL_X64` nor `DNNL_AARCH64` is ever defined.

So this is a breaking change which cannot be promoted as is. We have two options:
1. Either define the arch in `dnnl_config.h`
2. Or make X64 and AARCH64 isas disjoint and have all enumerations here.

Maybe we could (should?) do both :) 

Right now I am inclined towards 2nd option, but we need to seriously think about it.
Also tagging @nSircombe and @kruus.

Some relevant discussion here: https://github.com/oneapi-src/oneDNN/pull/682#issuecomment-610576147.

Reply:
It may be a temporary implementation, but I'm going to define both Intel and ARM as follows, how about this?
There are some minor variations in Arm ISAs, like AVX512_MIC/4OPS/VNNI, so it is expected that we will eventually run out of bit fields.

```c++
/// CPU instruction set flags
typedef enum {
    /// Any ISA (excepting those listed as initial support)
    dnnl_cpu_isa_all = 0x0,

    /// Intel Streaming SIMD Extensions 4.1 (Intel SSE4.1)
    dnnl_cpu_isa_sse41 = 0x1,
    ...
    /// Intel AVX-512, Intel DL Boost and bfloat16 support and
    /// Intel AMX with 8-bit integer and bfloat16 support
    /// (initial support)
    dnnl_cpu_isa_avx512_core_amx = 0x3e7,

    /// AARCH64 SIMD (128-bit instructions)
    dnnl_cpu_isa_simdfp = 0x1000000,

    /// AARCH64 SVE 128 bits
    dnnl_cpu_isa_sve_128 = 0x8000000,

    /// AARCH64 SVE 256 bits
    dnnl_cpu_isa_sve_256 = 0x18000000,

    /// AARCH64 SVE 384 bits
    dnnl_cpu_isa_sve_384 = 0x28000000,

    /// AARCH64 SVE 512 bits
    dnnl_cpu_isa_sve_512 = 0x38000000,

} dnnl_cpu_isa_t;
```

Reply:
Hmm, I thought I cannot continue this thread, so started a new one here: https://github.com/oneapi-src/oneDNN/pull/850#discussion_r508980999. I resolve this discussion in favor the one below.

Sorry for the mess.

Spoiler: I changed my mind and agreed with your original approach. However, we need to be cautious about bringing `DNNL_AARCH64/DNNL_X64` macro outside.